workspaces/src/types/mod.rs

```diff
@@ -28,25 +28,79 @@ pub type Balance = u128;
 /// Height of a specific block
 pub type BlockHeight = u64;
 
+/// Key types supported for either a [`SecretKey`] or [`PublicKey`]
+#[derive(Debug, Copy, Clone, Serialize, Deserialize)]
+#[non_exhaustive]
+pub enum KeyType {
+    ED25519,
+    SECP256K1,
+}
+
+impl From<KeyType> for near_crypto::KeyType {
+    fn from(key_type: KeyType) -> Self {
+        match key_type {
+            KeyType::ED25519 => Self::ED25519,
+            KeyType::SECP256K1 => Self::SECP256K1,
+        }
+    }
+}
+
+impl From<near_crypto::KeyType> for KeyType {
+    fn from(key_type: near_crypto::KeyType) -> Self {
+        match key_type {
+            near_crypto::KeyType::ED25519 => Self::ED25519,
+            near_crypto::KeyType::SECP256K1 => Self::SECP256K1,
+        }
+    }
+}
```

Comment:
Not 100% sure we want this impl to be public (for maintenance reasons) is there any reason someone might want to convert to this rather than just having it as a helper function?

Reply:
What you changed it to lgtm, we can always add it later, but a more breaking change to remove and this way the API is more streamlined and harder to misuse